lib/openshift/machine.rb

```diff
@@ -43,6 +43,34 @@ def ready?(user: nil, cached: true, quiet: false)
       instance_state == 'running'
     end
 
+    #azure location 
+    def azure_location(user: nil, cached: true, quiet: false)
+       raw_resource(user: user, cached: cached ,quiet: quiet).
+         dig('spec', 'providerSpec', 'value', 'location')
+    end
+
+    #gcp region and zone
+    def gcp_region(user: nil, cached: true, quiet: false)
+       raw_resource(user: user, cached: cached ,quiet: quiet).
+         dig('spec', 'providerSpec', 'value', 'region')
+    end
+
+    def gcp_zone(user: nil, cached: true, quiet: false)
+       raw_resource(user: user, cached: cached ,quiet: quiet).
+         dig('spec', 'providerSpec', 'value', 'zone')
+    end
+
+    #aws ami-id and availability zone
+    def aws_ami(user: nil, cached: true, quiet: false)
```

Comment:
this returns both - ami id and availability zone? This is not very good especially if method name is `aws_ami`. I think it should be `aws_ami_id` and return only AMI id, then have another one for zone. But if you want both, call it some meaningful name.

Reply:
ok, I see two different methods. Actually if method name is clear like `aws_ami_id`, adding a comment can only confuse the meaning. please remove the comments.